beetsplug/substitute.py

```diff
@@ -0,0 +1,40 @@
+# This file is part of beets.
+# Copyright 2016, Adrian Sampson.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+"""Uses user-specified substituting rules to canonicalize names for path
+formats without modifying the tags of the songs.
+"""
+from beets.plugins import BeetsPlugin
+import re
+
+_substitute_rules = []
```

Comment:
I'd recommend making this a field on the class. Then `_tmpl_substitute` could be a method as well and use `self.substitute_rules`.

Reply:
Fixed